script.js

```diff
@@ -1,41 +1,106 @@
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// Function to initialize the webpage
+function initializePage() {
+  fetchEpisodesData();
 }
 
-function makePageForEpisodes(episodeList) {
+// Fetch episode data from TVMaze API
+function fetchEpisodesData() {
+  const apiUrl = 'https://api.tvmaze.com/shows/82/episodes';
+
+  fetch(apiUrl)
+      .then(response => {
+          if (!response.ok) {
+              throw new Error('Network response was not ok');
+          }
+          return response.json();
+      })
+      .then(episodesData => {
+          renderEpisodes(episodesData);
+          addSearchEventListener(episodesData);
+      })
+      .catch(error => {
+          handleFetchError(error);
+      });
+}
+
+// Render episodes on the webpage
+function renderEpisodes(episodesData) {
   const rootElem = document.getElementById("root");
+  removeAllChildren(rootElem);
+
+  episodesData.forEach(episode => {
+      const episodeCard = createEpisodeCard(episode);
+      rootElem.appendChild(episodeCard);
+  });
 
-  episodeList.forEach(episode => {
-      const episodeContainer = document.createElement("div");
-      episodeContainer.classList.add("episode");
+  displayEpisodeCount(episodesData.length);
+}
+
+// Create an episode card
+function createEpisodeCard(episode) {
+  const episodeContainer = document.createElement("div");
+  episodeContainer.classList.add("episode");
 
-      // Here I created the episode code
-      const episodeCode = `S${pad(episode.season)}E${pad(episode.number)}`;
+  const episodeCode = `S${pad(episode.season)}E${pad(episode.number)}`;
 
-      // Then I created  the episode title
-      const titleElem = document.createElement("h2");
-      titleElem.textContent = `${episode.name} - ${episodeCode}`;
-      episodeContainer.appendChild(titleElem);
+  const titleElem = document.createElement("h2");
+  titleElem.textContent = `${episode.name} - ${episodeCode}`;
+  episodeContainer.appendChild(titleElem);
 
-      // Then I createa the  episode image
-      const imageElem = document.createElement("img");
-      imageElem.src = episode.image.medium;
-      imageElem.alt = `${episode.name} Image`;
-      episodeContainer.appendChild(imageElem);
+  const imageElem = document.createElement("img");
+  imageElem.src = episode.image.medium;
+  imageElem.alt = `${episode.name} Image`;
+  episodeContainer.appendChild(imageElem);
 
-      // after the I  created the  episode summary
-      const summaryElem = document.createElement("div");
-      summaryElem.innerHTML = episode.summary;
-      episodeContainer.appendChild(summaryElem);
+  const summaryElem = document.createElement("div");
+  summaryElem.innerHTML = episode.summary;
+  episodeContainer.appendChild(summaryElem);
 
-      rootElem.appendChild(episodeContainer);
+  return episodeContainer;
+}
+
+// Add event listener to search input for filtering episodes
```

Comment:
Generally, you wouldn't want your comments to describe what the code is doing, but rather why the code is doing what it is doing.